### PR TITLE
feat(icon): add input for inline styling of icons

### LINKED ISSUE #9742: feat(icon): create convenience input/attribute for inlining icons into text areas
As described.

### PR DESCRIPTION
**Pull request by josephperrott**
Fixes #9742

### CODE REVIEW

File: src/lib/icon/icon.ts [hunk 1]
```diff
@@ -67,13 +68,25 @@ export const _MatIconMixinBase = mixinColor(MatIconBase);
...
     'class': 'mat-icon',
+    '[class.mat-icon-inline]': 'inline',
   },
   encapsulation: ViewEncapsulation.None,
   preserveWhitespaces: false,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, CanColor {
 
+  /** Whether the icon should be inline. */
```
Comment (jelbourn) [inline src/lib/icon/icon.ts]: Expand comment to explain what "inline" means?

File: src/lib/icon/icon.ts [hunk 2]
```diff
@@ -67,13 +68,25 @@ export const _MatIconMixinBase = mixinColor(MatIconBase);
...
+  @Input()
+  get inline(): boolean {
+    return this._inline;
+  }
+  set inline(inline: boolean) {
+    this._inline = coerceBooleanProperty(inline);
+  }
+  private _inline: boolean = false;
```
Comment (jelbourn) [inline src/lib/icon/icon.ts]: Would be good to have one unit test that uses this.

File: src/lib/icon/icon.scss [hunk 3]
```diff
@@ -10,6 +10,13 @@ $mat-icon-size: 24px !default;
   fill: currentColor;
   height: $mat-icon-size;
   width: $mat-icon-size;
+
+  &.mat-icon-inline {
+    font-size: inherit;
```
Comment (jelbourn) [inline src/lib/icon/icon.scss]: Does IE11 support `inherit`? Any reason for not using `1em`?

Comment (donroyco) [reply src/lib/icon/icon.scss]: FYI: `inherit` [is  supported](https://developer.mozilla.org/en-US/docs/Web/CSS/inherit) since IE8.

Comment (jelbourn) [reply src/lib/icon/icon.scss]: Ah, I was thinking of `initial`

Comment (josephperrott) [reply src/lib/icon/icon.scss]: We could use `1em` instead of `inherit` if you would like, I don't have a super strong preference at this point.

Comment (jelbourn) [reply src/lib/icon/icon.scss]: `inherit` should be fine, I mixed up the IE support in my head

File: src/lib/icon/icon.spec.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-import {FAKE_SVGS} from './fake-svgs';
-import {wrappedErrorMessage} from '@angular/cdk/testing';
+import { inject, async, fakeAsync, tick, TestBed } from '@angular/core/testing';
+import { SafeResourceUrl, DomSanitizer } from '@angular/platform-browser';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { Component } from '@angular/core';
+import { MatIconModule } from './index';
+import { MatIconRegistry, getMatIconNoHttpProviderError } from './icon-registry';
+import { FAKE_SVGS } from './fake-svgs';
+import { wrappedErrorMessage } from '@angular/cdk/testing';
```
Comment (jelbourn) [inline src/lib/icon/icon.spec.ts]: Looks like your IDE added some spaces here and elsewhere. WebStorm lets you configure this, not sure about Code.

**Pull request merged**